llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2821,6 +2823,55 @@ class VPScalarIVStepsRecipe : public VPRecipeWithIRFlags {
   }
 };
 
+/// A recipe to represent scalar stores that sink outside the vector loop.
+class VPIntermediateStoreRecipe : public VPRecipeBase {
```

Comment:
Could this simply be a VPInstruction opcode?

Reply:
I believe the way Mel implemented it is similar to how `VPWidenStore` is done, with the difference that it's scalar store. Maybe proper name would be `VPScalarStore` though, like `VPScalarCast`

Reply:
In general, VPInstruction is intended to be used for cases where Opcode + VPVAlue operands are sufficient to define the operation to avoid explosion of recipe classes. There are other opcodes that don't define values (this is similar to LLVM IR's Instructions which also all are Values, but possibly of VoidTy.

In this case, it looks like it would be good to store additional data (alignment), so having a separate recipe seems fine.